src/Initializer/BatchRecorders/DynamicRuptureRecorder.cpp

```diff
@@ -0,0 +1,104 @@
+#include "Recorders.h"
+#include <Kernels/Interface.hpp>
+#include <yateto.h>
+
+using namespace device;
+using namespace seissol::initializers;
+using namespace seissol::initializers::recording;
+
+
+void DynamicRuptureRecorder::record(DynamicRupture &handler, Layer &layer) {
+  setUpContext(handler, layer);
+  recordDofsTimeEvaluation();
+  recordSpaceInterpolation();
+}
+
+
+void DynamicRuptureRecorder::recordDofsTimeEvaluation() {
+  real** timeDerivativePlus = currentLayer->var(currentHandler->timeDerivativePlus);
+  real** timeDerivativeMinus = currentLayer->var(currentHandler->timeDerivativeMinus);
+  real* idofsPlus = static_cast<real *>(currentLayer->getScratchpadMemory(currentHandler->idofsPlusOnDevice));
+  real* idofsMinus = static_cast<real *>(currentLayer->getScratchpadMemory(currentHandler->idofsMinusOnDevice));
+
+  if (currentLayer->getNumberOfCells()) {
+    std::vector<real *> timeDerivativePlusPtrs{};
+    std::vector<real *> timeDerivativeMinusPtrs{};
+    std::vector<real *> idofsPlusPtrs{};
+    std::vector<real *> idofsMinusPtrs{};
+
+    const size_t idofsSize = tensor::Q::size();
+    for (unsigned face = 0; face < currentLayer->getNumberOfCells(); ++face) {
+      timeDerivativePlusPtrs.push_back(timeDerivativePlus[face]);
+      timeDerivativeMinusPtrs.push_back(timeDerivativeMinus[face]);
+      idofsPlusPtrs.push_back(&idofsPlus[face * idofsSize]);
+      idofsMinusPtrs.push_back(&idofsMinus[face * idofsSize]);
+    }
```

Comment:
Relevant only if this thing here is called often (and not e.g. in init):
You could use reserve(size) on the vector or allocate it at once. Would save some re-allocations. If it isn't in a relevant loop, ignore this :)

Reply:
Yes, these loop are getting executed only once in the initialization stage. I agree about this particular loop. However, there are some places where I don't know sizes in advance.

Reply:
Okay, if they are only in the init that's fine.